pandas/io/formats/style.py

```diff
@@ -1028,14 +1046,18 @@ def _background_gradient(s, cmap="PuBu", low=0, high=0, text_color_threshold=0.4
             raise ValueError(msg)
 
         with _mpl(Styler.background_gradient) as (plt, colors):
-            smin = s.values.min()
-            smax = s.values.max()
+            smin = s.min() if vmin is None else vmin
+            if isinstance(smin, ABCSeries):
```

Comment:
Is this not already covered by the line directly preceding it?

Reply:
Here depends on the `axis` argument provided to `Styler.background_gradient`, `s` can be a DataFrame or Series. if `s` is a DataFrame and `vmin` not provided, smin will first become a Series then become the min value of the full data, the code here is consistent with (aka. direct copy from) `Styler.bar` method, do we want to change it?

Reply:
side note: try to use `s.values.min()` to get overall minima will produce `AttributeError: 'IntegerArray' object has no attribute 'min'` as described in #28869. OTOH, use `np.min(s.values)` will also produce `NotImplementedError: The 'reduce' method is not supported.` I thought IntegerArray might be a WIP.

Reply:
Can you move this into a helper function? I think this affects `highlight_extrema` as well but seems to tackle slightly differently; should keep unified

`squeeze` might also be more appropriate to use here

Reply:
Thanks! didn't notice the `highlight_extrema`, they really should be consistent. turned-out a fairly simple approach can be used: `np.nanmin(s.to_numpy())`